include/swift/SILOptimizer/Utils/BasicBlockOptUtils.h

```diff
@@ -0,0 +1,260 @@
+//===--- BasicBlockOptUtils.h - SIL basic block utilities -------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+///
+/// Utilities used by the SILOptimizer for analyzing and operating on whole
+/// basic blocks, including as removal, cloning, and SSA update.
+///
+/// CFGOptUtils.h provides lower-level CFG branch and edge utilities.
+///
+/// SIL/BasicBlockUtils.h provides essential SILBasicBlock utilities.
+///
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_SILOPTIMIZER_UTILS_BASICBLOCKOPTUTILS_H
+#define SWIFT_SILOPTIMIZER_UTILS_BASICBLOCKOPTUTILS_H
+
+#include "swift/SIL/SILInstruction.h"
+#include "swift/SIL/SILBasicBlock.h"
+#include "swift/SIL/SILCloner.h"
+
+namespace swift {
+
+class BasicBlockCloner;
+class SILLoop;
+class SILLoopInfo;
+
+/// Remove all instructions in the body of \p BB in safe manner by using
+/// undef.
+void clearBlockBody(SILBasicBlock *BB);
+
+/// Handle the mechanical aspects of removing an unreachable block.
+void removeDeadBlock(SILBasicBlock *BB);
+
+/// Remove all unreachable blocks in a function.
+bool removeUnreachableBlocks(SILFunction &Fn);
+
+/// Return true if there are any users of V outside the specified block.
+inline bool isUsedOutsideOfBlock(SILValue V) {
+  auto *BB = V->getParentBlock();
```

Comment:
As an example of where the style is no beuano b/c of the age of the code.

Reply:
I suppose I could make a pass over the code and convert variable names to lower camel. Are there any other specific conventions that you think need to be fixed? I don't want to rewrite anything in this PR.

Reply:
I am fine with that and would like a git-clang-format as well. I would do it as a separate commit for cleanliness purposes. (I don't think it needs to be a separate PR, but you do you?).